### DISCUSSION TITLE
Azure blob storage sink error - no key/value found in connection string

### DISCUSSION THREAD

**Discussion by ubergeekNZ** [vectordotdev/vector#15170] [Q&A] [answered]
Hi,

I am trying to setup vector to send the log to Azure Storage bucket but getting the following error.

`2022-11-10T08:27:25.389600Z ERROR vector::topology: Configuration error. error=Sink "my_sink_id": no key/value found in connection string: **REDACTED**
`
My configuration for the sink is shown below (copied from the documentation).

`[sinks.my_sink_id]
type = "azure_blob"
inputs = [ "parse_logs"]
connection_string = "DefaultEndpointsProtocol=https;AccountName=**<account_name>**;AccountKey=**<account_key>**;EndpointSuffix=core.windows.net"
container_name = "device_logs"
blob_prefix = "raspberry01blob/%F/"
blob_append_uuid = true
compression = "gzip"
blob_time_format = "%s"

  [sinks.my_sink_id.encoding]
  codec = "json"
`

**Comment by jszwedko** [comment, marked as answer]
Hi @ubergeekNZ ! One thought: did you base64 encode the account key?

**Reply by ubergeekNZ** [reply]
yeah I tried it using python

https://stackabuse.com/encoding-and-decoding-base64-strings-in-python/

But still the same error.

**Reply by jszwedko** [reply]
This error seems to be coming from the Azure SDK: https://github.com/Azure/azure-sdk-for-rust/blob/b4544d4920fa3064eb921340054cd9cc130b7664/sdk/storage/src/connection_string.rs#L115 . Do you have any stray `=` characters outside of key/value? For example, it seems like you might want to drop any `=` from the base64 padding if they are there.

**Reply by archoversight** [reply]
@jszwedko it should be splitting on the first `=` found, which should then return a key/value pair. The extra `=`'s in the value shouldn't matter.

That being said, I am here cause I am seeing the same issue.

**Reply by archoversight** [reply]
https://github.com/Azure/azure-sdk-for-rust/pull/622/files fixed the issue of splitting on the first `=` since that was required to fix https://github.com/vectordotdev/vector/issues/10979 ;-)

**Reply by archoversight** [reply]
This is an issue with a change that was introduced in https://github.com/vectordotdev/vector/pull/14305 and I have provided a fix in: https://github.com/vectordotdev/vector/pull/15225
